### PR TITLE
Fix importing ics files on Windows

### LINKED ISSUE #1898: No calendar import
### Steps to reproduce

1. Download the calendar from the old version 1.7.1
2. Open the calendar settings of version 2.0.1
3. Import calendar copy file.
4. **No import action.**

### Expected behaviour
Import calendar from file.

### Actual behaviour
No display of tasks from calendar import from file.

### Calendar app
**Calendar app version:** (see apps admin page, e.g. 2.0.1)
2.0.1

### Server configuration
**Nextcloud Version:** (see admin page, e.g. 17.0.2)
18.0.0

### LINKED ISSUE #1920: Better feedback for import failures
Proper feedback when parsing fails, file is not actually a text/calendar, etc.

### PR DESCRIPTION
**Pull request by georgehrke**
fixes #1898 
fixes #1920

### CODE REVIEW

File: src/components/AppNavigation/Settings/SettingsImportSection.vue [hunk 1]
```diff
@@ -152,11 +151,32 @@ export default {
...
+				// "Uncommon" file-extensions will result in an empty type
+				// and apparently Microsoft considers calendar files to be "uncommon"
+				if (type === '') {
+					// If it's an xml file, our best guess is xCal: https://tools.ietf.org/html/rfc6321
+					// If it's a json file, our best guess is jCal: https://tools.ietf.org/html/rfc7265
+					// In every other case, our best guess is just plain old iCalendar: https://tools.ietf.org/html/rfc5545
+					if (name.endsWith('.xml')) {
+						type = 'application/calendar+xml'
+					} else if (name.endsWith('.json')) {
+						type = 'application/calendar+json'
```
Comment (tcitworld) [inline src/components/AppNavigation/Settings/SettingsImportSection.vue]: `JSCalendar` will be an issue when it comes around.

File: src/components/AppNavigation/Settings/SettingsImportSection.vue [hunk 2]
```diff
@@ -152,11 +151,32 @@ export default {
...
+					} else {
```
Comment (tcitworld) [inline src/components/AppNavigation/Settings/SettingsImportSection.vue]: Maybe add `.csv` :arrow_right:  `text/csv` as well ?

File: src/components/AppNavigation/Settings/SettingsImportSection.vue (hunk 1 above)
Comment (georgehrke) [reply src/components/AppNavigation/Settings/SettingsImportSection.vue]: Hm, Indeed. But as long as `JSCalendar` is still a draft and as long as we only support jCal anyway, i would not worry about it for now.

File: src/components/AppNavigation/Settings/SettingsImportSection.vue (hunk 2 above)
Comment (georgehrke) [reply src/components/AppNavigation/Settings/SettingsImportSection.vue]: Good point, let me add that

**Pull request merged**